# LIHC_protein_modeling-
  ###  ðŸ§¬  Protein Interaction Prediction in Hepatocellular Carcinoma using Transcriptomics and Functional Similarity.


  
  ----- 
  ####  ðŸš€ Motivation :


This project was based on the idea that transcriptomic data can indirectly reflect functional protein associations, especially when combined with biological context such as Gene Ontology (GO) similarity and prior knowledge from STRINGdb. By leveraging gene co-expression and semantic similarity, we aimed to build a predictive model capable of identifying potential protein-protein interactions (PPIs) specific to liver cancer biology. This model could help narrow down candidate interactions for wet-lab validation.


  
  ----
  ####  ðŸ’» Methods:

we first downloaded RNA-Seq data from TCGA-LIHC (hepatocellular carcinoma) and identified differentially expressed genes (DEGs) using DESeq2. We applied strict thresholds (padj < 0.05, |log2FoldChange| > 1) for significant genes. Next, we extracted protein-protein interaction data from STRINGdb (v11.5, Homo sapiens) and filtered it down to high-confidence interactions. Starting with over 517146  pairs (combined score > 700), we tightened the criteria to just 3,314 pairs (combined score > 900) to focus on the strongest signals.

To assign biolgically meaningful labels, we calculated Gene Ontology (GO)similarity (Biological Process ontology) for all pairs using GOSemSim package. Pairs with go_sim >= 0.6 were labeld as interacting (1) while those with go_sim <= 0.4 were label as non-interacting (0), pairs with intermediate similarity (0.4 < go_sim < 0.6) were excluded to minimize ambiguity leaving 1378 pairs for further analysis.

For each PPI we computed CO-expression (cor) Pearson correlation of RNA-seq expression levels ,which was calculated only on training data to prevent data leakage in modeling. We trained a logistic regression model (glm) to predict interactions. Co-expression correlation and combined-score  were used as predictors while binary lables ((1) = 519 /(0) = 859) from GO semantics used as response.

Finall Model was evaluted by both biological validation and Independent test set (30% held-out data).Gene Set enrichmet analysis (GSEA) of top-predicted genes was conducted by clusterProfiler package in R 

____
####  ðŸ Results :

From the LIHC RNA-seq data, we identified 1232 significant differentially expressed genes using Desq2, these genes served as the candidat set for potential PPIs. A heatmap of top 50 significant genes are illustreted below
<img width="1181" height="682" alt="HEATMAP" src="https://github.com/user-attachments/assets/3b4370c0-28d1-4ae1-8f09-b42bffe21ed8" />

____

#### ðŸ“¦ packages used :


-----

#### ðŸ¥‡ future enhancments:
